x-pack/plugins/case/server/routes/api/__tests__/delete_case.test.ts

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  mockCases,
+  mockCasesErrorTriggerData,
+  createMockSavedObjectsRepository,
+  createRouteContext,
+} from '../__fixtures__';
+import { initDeleteCaseApi } from '../delete_case';
+import { IRouter, kibanaResponseFactory } from 'src/core/server';
+import { loggingServiceMock, httpServiceMock, httpServerMock } from 'src/core/server/mocks';
+import { CaseService } from '../../../services';
+import { securityMock } from '../../../../../security/server/mocks';
+
+describe('DELETE case', () => {
+  const setup = async () => {
+    const httpService = httpServiceMock.createSetupContract();
+    const router = httpService.createRouter('') as jest.Mocked<IRouter>;
+
+    const log = loggingServiceMock.create().get('case');
+
+    const service = new CaseService(log);
+    const caseService = await service.setup({
+      authentication: securityMock.createSetup().authc,
+    });
+
+    initDeleteCaseApi({
+      router,
+      caseService,
+    });
+
+    return {
+      routeHandler: router.delete.mock.calls[0][1],
+    };
+  };
+  it(`deletes the case. responds with 204`, async () => {
+    const { routeHandler } = await setup();
```

Comment:
is `routeHandler` belong inside of `beforeAll` function

Reply:
in some cases, these take arguments that trigger the mocked saved object api to throw errors. ill clean it up though. thanks